### PR TITLE
Imporve version resolution logic

### PR DESCRIPTION
**Pull request by donchev7**
First stab at [issue 13](https://github.com/shabanzd/dependency-bazelizer/issues/13)

### CODE REVIEW

**Review by shabanzd — COMMENTED** [review]
Running :
` bazelisk run //src:dependency-bazelizer -- -if {your_path}/src/deb_packages.in`
yields:
`  File "/home/varvara/.cache/bazel/_bazel_varvara/1c3e3ba2b0b3f2cb363c4f67a0d29b52/execroot/_main/bazel-out/k8-fastbuild/bin/src/dependency-bazelizer.runfiles/rules_python~0.21.0~pip~pip_packaging/site-packages/packaging/version.py", line 198, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
packaging.version.InvalidVersion: Invalid version: '5.15.0-1ubuntu2'`

**Review by shabanzd — COMMENTED** [review]
I am not sure packaging is the correct lib to handle Debian packages versions ...
This is the spec they parse versions with respect to: https://peps.python.org/pep-0440/
This is what we need : https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-version
The python-apt package looks interesting

**Comment by donchev7** [comment]
**python-apt** is not used for parsing version strings. 
Its primary use is to interface with the apt package management system.

I'll create a custom function for now. When we find a debian version parser we can re-evaluate.

**Comment by shabanzd** [comment]
image

We still have complaints at line 89 about the version. Although it is a compliant debian version

**Comment by donchev7** [comment]
Now it correctly parses debian versions and Dependents symbols (>= 1. ....etc..)

File: tests/BUILD.bazel [hunk 1]
```diff
@@ -10,6 +10,15 @@ py_test(
     ],
 )
 
+py_test(
+    name = "test_get_package_version",
+    timeout = "short",
+    srcs = ["test_get_package_version.py"],
+    deps = [
+        "//src:dependency-bazelizer",
```
Comment (shabanzd) [inline tests/BUILD.bazel:18]: you don't need the entire dependency-bazelizer, src:get_package_version should be enough

File: src/get_package_version.py [hunk 2]
```diff
@@ -1,19 +1,83 @@
...
+
+    # Process debian_version to make it compatible with packaging.version.parse
+    debian_version = debian_version.split('~')[0]
+    debian_version = debian_version.split('+')[0]
+    debian_version = debian_version.split('-')[0]
+    debian_version = re.split('[^0-9.]', debian_version)[0].rstrip('.')
+
+    return epoch, debian_version
+
+def compare_debian_versions(version1: str, version2: str) -> int:
```
Comment (shabanzd) [inline src/get_package_version.py:55]: maybe return an enum here ?
Like LogicalOperator enum that has LARGER_THAN, ... etc

File: src/get_package_version.py [hunk 3]
```diff
@@ -1,19 +1,83 @@
...
+    epoch2, debian_version2 = parse_debian_version(version2)
+
+    # Compare epochs as integers if they are both present
+    if epoch1 is not None and epoch2 is not None:
+        if epoch1 < epoch2:
+            return -1
+        elif epoch1 > epoch2:
+            return 1
+    
+    # The epoch is a number that can be used to ensure that all later versions of the package are considered "newer" than all earlier versions.
```
Comment (shabanzd) [inline src/get_package_version.py:66]: I think this comment belongs earlier in the function

File: src/get_package_version.py [hunk 4]
```diff
@@ -50,45 +114,41 @@ def _get_deb_package_version_from_aptcache(name: str, arch: str) -> str:
...
-            else ""
-        )
 
-    versions = [
-        version.name
-        for version in Path.iterdir(module_path)
-        if Path.is_dir(Path.joinpath(module_path, version))
-    ]
+    versions =  fs.get_directories(module_path)
+
```
Comment (shabanzd) [inline src/get_package_version.py:130]: Opinion: unnecessary empty line

**Review by shabanzd — APPROVED** [review]

File: src/get_package_version.py (hunk 3 above)
Comment (donchev7) [reply src/get_package_version.py]: done

File: src/get_package_version.py (hunk 4 above)
Comment (donchev7) [reply src/get_package_version.py]: removed

File: tests/BUILD.bazel (hunk 1 above)
Comment (donchev7) [reply tests/BUILD.bazel]: thx

**Pull request merged**